NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# Edit comments

### Description
Added functionality to delete a comment that the user created but cannot delete other users' comments
[Trello Ticket 22](https://trello.com/c/TAOYqDD3/22-22-edit-a-comment)


### Testing Instructions
1. `git add` and `git commit` whatever you're working on in your branch
2. `git checkout main`
3. `git fetch -a`
4. `git checkout lv-edit-comment`
5. If you don't see the most recent version of my files `git pull origin lv-edit-comment`
6. If you have not created the database please do so and follow Steps 7-8 otherwise proceed to Step 9
7. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/01_Tabloid_Create_DB.sql) to create the initial database
8. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/02_Tabloid_Seed_Data.sql) to seed the initial data
9. Open the TabloidFullStack.sln with Visual Studio
10. At the top of Visual Studio, select https as the launcher and click the green play button
11. The Swagger UI should appear in the browser
12. Confirm there is a PUT call under the Comment Controller in Swagger
13. If the React app isn't already running, in your terminal `npm start` otherwise refresh the React App
14. 19. Once the app is up and running, login with `foo@bar.com` and no password
20. Select `Posts`
21. Click on the post title: `grow frictionless e-tailers`
22. You should be redirected to **/post/16**
23. Select `Add Comment`
24. You should be redirected to **/post/16/Comments/Add**
25. Confirm `Add a new comment to post: grow frictionless e-trailers` is at the top of the page
26. Enter `Test` in the subject input field
27. Enter `Test Test Test` in the content textarea
28. Select Save
29. You should be redirected to **post/16/Comments**
30. Confirm that the new comment is at the top of the list
31. Confirm that the new comment has a grey `Edit` button and the other comment by rdo1 does not.
32. Click the grey `Edit` button
33. You should be redirected to **/post/16/Comments/Edit/{commentId}**
34. Confirm you see the following:
![EditComment](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/assets/77558296/e079bcf0-1030-42e1-9f43-ff25360ab883)
35. Change the subject to `Test 2`
36. Change the content to `Test 1 Test 2 Test 3`
37. Click Save
38. You should be redirected  to the comment details page at **http://localhost:3000/Post/16/Comments/{commendId}**
39. Confirm the changes you made are displayed in the comment
40. Select `Back to comments`
41. You should be redirected to **post/16/Comments**
42. Select the grey `Edit` button again
43. Select the grey `Cancel` button
44. You should be redirected to **post/16/Comments**
45. Confirm that none of the information has been changed on your comment
46. enter **http://localhost:3000/post/16/Comments/Edit/21** in the address bar
47. Confirm you see the following message
> Must be the author of the comment to edit the comment
48. Select `back to comments`
49.  You should be redirected to **post/16/Comments**

### Checklist
